Dust

zzaro opened the pull request:
RestTemplate을 써봤는데 어째서인지 오류가..

hisomini committed:
#5 학식정보 크롤링

hisomini committed:
#5 학식정보 크롤링

tuguri8 committed:
Merge pull request #7 from Parksomini/develop

#5 학식정보 크롤링

zzaro committed:
Add files via upload

일단 미세먼지 JSON 파싱까지만 구현

zzaro committed:
GachonController를 RestTemplate으로 수정

zzaro committed:
RestTemplate으로 수정

tuguri8 commented:
일단 패키지 위치가 잘못되어 있는 것 같습니다. 
별도로 ``gachon-develop``이라는 패키지를 만드실 필요가 없어요 !

수정후에 **develop**으로 PR 날려주시면 감사하겠습니다!

tuguri8 commented on gachonbot-develop/src/main/java/com/bot/gachon/controller/GachonController.java:
```diff
@@ -0,0 +1,50 @@
+    @GetMapping("/dust")
+    public String callAPI() throws JsonProcessingException {
+
+        HashMap<String, Object> result = new HashMap<String, Object>();
```
Spring Boot는 자동으로 JSON변환을 해주는 ``Jackson`` 라이브러리를 내장하고 있기 때문에, 별도로 ``Map``을 선언해서 넣어주기 보다는, 클래스를 정의해서 객체에 넣어주면 됩니다.

tuguri8 commented on gachonbot-develop/src/main/java/com/bot/gachon/controller/GachonController.java:
```diff
@@ -0,0 +1,50 @@
+            UriComponents uri = UriComponentsBuilder.fromHttpUrl(url+"?"+"serviceKey=%2FFMFpVx%2Bvq%2BeEKK6owg7QDf44oVwyYvRL8I8DAo1ZgHpJmeQ0YXKYlelKIafadOJJ5nB%2B1OV1l5pJyKwMuI3Pg%3D%3D&numOfRows=10&pageNo=1&sidoName=%EC%84%9C%EC%9A%B8&searchCondition=DAILY&_returnType=json").build();
+
+            //API 호출
+            ResponseEntity<Map> resultMap = restTemplate.exchange(uri.toString(), HttpMethod.GET, entity, Map.class);
```
RestTemplate 사용법은 아래 주소에서 참고하시면 될 것 같아요!
https://sooin01.tistory.com/entry/RestTemplate-%EC%82%AC%EC%9A%A9%EB%B2%95

zzaro replied:
public SampleResponse doSample(String userId) {
        RestTemplate restTemplate = new RestTemplate();
        URI uri = UriComponentsBuilder.fromHttpUrl("http://openapi.airkorea.or.kr/openapi/services/rest/ArpltnInforInqireSvc/getCtprvnMesureSidoLIst?serviceKey=P4zgBHeiRu9%2B499IV12fo7S463t6lMsVwSGZqk1mVWYVzNnpcGrp8MMCwZg2R6cmBiT4eSqmdj%2BrnAl59wkdJg%3D%3D&numOfRows=10&pageNo=1&sidoName=%EC%84%9C%EC%9A%B8&searchCondition=DAILY&_returnType=json") .queryParam("key", "value") .build().toUri();
        ResponseEntity<String> responseEntity = restTemplate.getForEntity(uri, String.class);

        return null;
    }

RestTemplate은 이것만으로 구현이 끝나는거고 나머지는 MVC 패턴을 구현하면 되는건가요? 결과값을 출력해서 눈으로 확인이 안되니 정상적으로 구현된건지 긴가민가하네요..

tuguri8 replied:
``http://openapi.airkorea.or.kr/openapi/services/rest/ArpltnInforInqireSvc/getCtprvnMesureSidoLIst?serviceKey=P4zgBHeiRu9%2B499IV12fo7S463t6lMsVwSGZqk1mVWYVzNnpcGrp8MMCwZg2R6cmBiT4eSqmdj%2BrnAl59wkdJg%3D%3D&numOfRows=10&pageNo=1&sidoName=%EC%84%9C%EC%9A%B8&searchCondition=DAILY&_returnType=json``

해당 주소로 요청을 보내면 ``JSON``형식으로 응답을 주는걸로 알고 있는데요,
앞서 말했듯이 Spring Boot는 자동으로 ``JSON``형식을 변환을 해주는 ``Jackson`` 라이브러리를 내장하고 있기 때문에, 해당 응답에 맞는 객체를 만들어서 넣어주면 됩니다.

위에서는 ``String``으로 받는걸로 되어있지만, 해당 응답형식에 맞는 클래스를 선언해서 넣어주면 잘 들어가겠네요.

tuguri8 replied:
``REST API``를 개발하는 것이기 때문에 뷰는 따로 개발할 필요가 없습니다.
보통 기본적으로 ``Controller`` - ``Service`` - 등등 으로 레이어를 나눠서 개발을 하는데,
간단히 지금 같은경우에는 ``Controller``에서는 ``Request``만 받고, ``Service``로 넘겨줘서 ``Service``에서 ``RestTemplate``을 통해 로직을 처리한다고 보면 됩니다!

hisomini committed:
#5 오늘의 학식

hisomini committed:
#5 오늘의 학식

hisomini committed:
#5 오늘의 학식(enum)

hisomini committed:
#5 오늘의 학식

tuguri8 committed:
Merge pull request #11 from gachonbot/somin

오늘의 학식

hisomini committed:
#6

tuguri8 committed:
Fix MaskDto

zzaro committed:
2번째 수정

"SERVICE KEY IS NOT REGISTERED ERROR" 오류가 발생하고 있습니다. 

일단 조치해본 사항으로는 공공데이터 API 서비스키는 이미 인코딩되어있기 때문에 decode해서 API를 날리고, URIComponents builder의 자동 인코딩을 막기 위해 build(false)를 넣었습니다.

hisomini committed:
#마스크정보 연동

hisomini committed:
Merge remote-tracking branch 'origin/somin' into somin

# Conflicts:
#	src/main/java/com/bot/gachon/dto/response/MaskDto.java
#	src/main/java/com/bot/gachon/dto/response/MaskSubDto.java

tuguri8 committed:
#2 refactor findDust

hisomini committed:
#마스크정보 연동

hisomini committed:
Merge pull request #13 from gachonbot/somin

#마스크정보 연동

hisomini committed:
#version

hisomini committed:
#마스크정보 업데이트

hisomini committed:
#마스크정보 업데이트

tuguri8 committed:
Merge pull request #14 from gachonbot/somin

#마스크정보 업데이트

tuguri8 committed:
Change Table Name, Add Index

tuguri8 committed:
여유 마스크 API 추가

hisomini committed:
마스크입고시간조회 API 추가

hisomini committed:
마스크입고시간조회 API 추가

tuguri8 committed:
Merge pull request #15 from gachonbot/somin

마스크입고시간조회 API 추가

zzaro committed:
#3 refactor

zzaro committed:
WeatherDto와 유사하게 만듦.

tuguri8 committed:
test

zzaro committed:
시군구별 -> 측정소별

zzaro committed:
Update Url.java

zzaro committed:
미세먼지 수치별 등급 구현

zzaro committed:
URL 오타 수정

tuguri8 committed:
Merge branch 'develop' of https://github.com/tuguri8/gachonbot into dust

# Conflicts:
#	src/main/java/com/bot/gachon/config/RestConfig.java
#	src/main/java/com/bot/gachon/controller/GachonController.java
#	src/main/java/com/bot/gachon/service/GachonService.java
#	src/main/java/com/bot/gachon/service/Url.java

zzaro committed:
JpaRepository 구현

zzaro committed:
JpaRepository 구현

zzaro committed:
Update DustRepository.java

zzaro committed:
불필요한 코드(학식, 마스크, 날씨) 삭제 + DB 연동 기능 구현중

zzaro committed:
DB 저장 기능 구현중

zzaro committed:
Update Dust.java

zzaro committed:
Update Dust.java

zzaro committed:
dust refactor v3

zzaro committed:
미세먼지 최종(DB 연동 + 등급 구현)
